server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2534,12 +2534,18 @@ public void openEngineAndSkipTranslogRecoveryFromSnapshot() throws IOException {
         assert routingEntry().recoverySource().getType() == RecoverySource.Type.SNAPSHOT : "not a snapshot recovery ["
             + routingEntry()
             + "]";
+        openEngineAndSkipTranslogRecovery(false);
+    }
+
+    void openEngineAndSkipTranslogRecovery(boolean syncFromRemote) throws IOException {
```

Comment:
This method having the same name as the method on line 2525, but somewhat unrelated behavior, is weird.

I don't have a great name idea.

Maybe you could make the other one call this one with explicit `true` for `syncFromRemote`? But that one seems to assume the currentStage is `PEER`.

Reply:
If you move the lines:

```
        recoveryState.validateCurrentStage(RecoveryState.Stage.INDEX);
        maybeCheckIndex();
```

into `openEngineAndSkipTranslogRecoveryFromSnapshot`, then the arg-less one could call this one with `true`.

Reply:
I cleaned this up as suggested and updated to use `openEngineAndSkipTranslogRecoveryFromSnapshot` for the search replica case.  I don't love the method name but it has all the steps we need.